### ISSUE TITLE
Color last samples differently to improve visbility when overlapping with previous path

### ISSUE THREAD

**Issue by kodfodrasz** [y20k/trackbook#95]
Hi!

I find Trackbook really nice, because it is so clean, polished and minimal.

I have a single really annoying inconvenience: when I'm passing over an already visited section on the trip I cannot see the current location, as all samples have the same color. (eg. returning from a peak on a trip) If the last sample could be distinguished by different color, or different sized marker (maybe both) that would improve greatly upon this. If the last few samples could also be distinguished (maybe a gradient from the "old" sample color to the "fresh" sample color over the last 3-5 samples) that would also improve the sense of direction.

Greetings!

**Comment by Hob111** [comment]
I would certainly find it helpful if the starting point of a track was in a different colour and the current point or finish point (for a saved track) a third colour. Is this possible?

**Comment by y20k** [comment]
What about adding a different/special marker either at the start of a track or at the end - instead of using color here. And: What should such a marker look like? A flag or an X or something else?

**Comment by kodfodrasz** [comment]
Those could also work. I also check the path sometimes while still tracking. The end (last sample?) would still be "flagged" this way while still tracking?

**Comment by y20k** [comment]
Hi @kodfodrasz & @Hob111 

while tracking the current position is already marked with a semi-transparent ring.

<a href="https://user-images.githubusercontent.com/9103935/110019190-72049980-7d28-11eb-8a55-0ceadde61097.png"><img src="https://user-images.githubusercontent.com/9103935/110019190-72049980-7d28-11eb-8a55-0ceadde61097.png" width="260" /></a>

For the flag idea I wrote about earlier. I started Inkscape and combined a flag Icon with the standard location dot.
<a href="https://user-images.githubusercontent.com/9103935/110019683-0838bf80-7d29-11eb-9ea7-ca0c2d6111e6.png"><img src="https://user-images.githubusercontent.com/9103935/110019683-0838bf80-7d29-11eb-9ea7-ca0c2d6111e6.png" width="278" /></a>
<a href="https://user-images.githubusercontent.com/9103935/110019786-22729d80-7d29-11eb-9912-6019e5eb7cd2.png"><img src="https://user-images.githubusercontent.com/9103935/110019786-22729d80-7d29-11eb-9912-6019e5eb7cd2.png" width="278" /></a>
<a href="https://user-images.githubusercontent.com/9103935/110019850-2e5e5f80-7d29-11eb-8467-04dc17a5e314.png"><img src="https://user-images.githubusercontent.com/9103935/110019850-2e5e5f80-7d29-11eb-8467-04dc17a5e314.png" width="278" /></a>

The result is already pretty decent and "in character" with the style of the app.
<a href="https://user-images.githubusercontent.com/9103935/110020037-649bdf00-7d29-11eb-8603-50503f8d6925.png"><img src="https://user-images.githubusercontent.com/9103935/110020037-649bdf00-7d29-11eb-8603-50503f8d6925.png" width="260" /></a>

**Comment by Hob111** [comment]
I think the flag is better than the semitransparent ring. Presumably it will show on a saved track as well as a currently recording track. Is there any chance of doing something similar for the starting point? I'm particularly concerned about saved tracks rather than a track that's being recorded.

**Comment by y20k** [comment]
@Hob111 I thought about adding a special marker for the start of a track, too. But I could not think of a good icon. Any ideas? Having just one icon - the flag - seems to me less ambiguous, than having the flag and another unclear icon.

**Comment by Hob111** [comment]
@y20k I agree that two different markers would be messy. You seem to have dismissed colour as a way of differentiating start and finish, is this difficult to implement using osmdroid (which I think you use)? The symbol/colours commonly used are a green pin for the start and a red pin for the finish. However, logically, if the start is marked the finish is implied, or vice versa. Since a pin is more commonly used than a flag how about a pin for the latest or final point? Coloured red if possible but not essential.

**Comment by y20k** [comment]
As a test I am using a red pin for the end of the track right now.

<a href="https://user-images.githubusercontent.com/9103935/110333835-5e0bb100-8022-11eb-8241-ed3f8ff9f40e.png"><img src="https://user-images.githubusercontent.com/9103935/110333835-5e0bb100-8022-11eb-8241-ed3f8ff9f40e.png" width="270">

**Comment by Hob111** [comment]
Looks good. I assume it would also show in saved tracks? 

What do you think @kodfodrasz?

**Comment by y20k** [comment]
> I assume it would also show in saved tracks?

I the current version the pin is only used in saved tracks. I did not touch the recording screen ("Map") so far.

**Comment by kodfodrasz** [comment]
I like it. Actually whichever version gets implemented does not matter for me, as any of them solves my pain point.

**Comment by y20k** [comment]
I tested the implementation using real recording on my phone. In my opinion the pin icon is too small. And if the track is longer a start-pin might be better. I need to fiddle a bit more around before I feel I got it right.

**Comment by y20k** [comment]
icons, colors and sizes are now to my liking - at least for this evening.

<a href="https://user-images.githubusercontent.com/9103935/110539745-eb83f980-8125-11eb-94b9-1b373b87f5a2.png"><img src="https://user-images.githubusercontent.com/9103935/110539745-eb83f980-8125-11eb-94b9-1b373b87f5a2.png" width="270" /></a>

**Comment by Hob111** [comment]
Looks really good to me.  :thumbsup:
Should save me the occasional frustration when I open an old track of a circular walk and can't remember where I started!
